CRM/Custom/Page/Field.php

```diff
@@ -133,27 +133,9 @@ public function browse() {
         $action -= CRM_Core_Action::DISABLE;
       }
 
-      switch ($customFieldBAO->data_type) {
-        case "String":
-        case "Int":
-        case "Float":
-        case "Money":
-          // if Multi Select field is selected in custom field
-          if ($customFieldBAO->html_type == 'Text') {
-            $action -= CRM_Core_Action::BROWSE;
-          }
-          break;
-
-        case "ContactReference":
-        case "Memo":
-        case "Date":
-        case "Boolean":
-        case "StateProvince":
-        case "Country":
-        case "File":
-        case "Link":
-          $action -= CRM_Core_Action::BROWSE;
-          break;
+      // Remove link to edit option group if there isn't one
+      if (!$customFieldBAO->option_group_id) {
+        $action -= CRM_Core_Action::BROWSE;
```

Comment:
This absolutely takes the prize for the most bizarre waste of 20 lines I've ever seen.

Reply:
🎉 🎉 **Why I'm awarding this one top prize** 🏆 🥇 🏆 
1. Switch statement: why even... dear god why?
2. The code comment made absolutely no sense.
3. The rest of the switch statement: LOL
4. None of it had anything to do with the actual criteria that needed to be checked